ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot

```diff
@@ -0,0 +1,77 @@
+    Run Kueue E2E Test    e2e_test.go
+
+Run Visibility test
+    [Documentation]    Run ginkgo visibilty single cluster test
+    [Tags]  Kueue
+    ...     DistributedWorkloads
+
+    Enable Visibility Feature Gate
+    Run Kueue E2E Test    visibility_test.go
+
+
+*** Keywords ***
+Prepare Kueue E2E Test Suite
+    [Documentation]    Prepare Kueue E2E Test Suite
+    ${result} =    Run Process    git clone -b ${KUEUE_REPO_BRANCH} ${KUEUE_REPO_URL} ${KUEUE_DIR}
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL    Unable to clone kueue repo ${KUEUE_REPO_URL}:${KUEUE_REPO_BRANCH}:${KUEUE_DIR}
+    END
+
+    Log To Console    Install the latest development version of kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl apply --server-side -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=Error detected while installing kueue
+
+    # Add label instance-type=on-demand on worker node
+    ${result} =    Run Process    oc get nodes -o name | tail -n1
+    ...    shell=true    stderr=STDOUT
+    Set Suite Variable    ${WORKER_NODE}    ${result.stdout}
+    ${return_code} =    Run And Return Rc    oc label ${WORKER_NODE} instance-type=on-demand
+    Should Be Equal As Integers  ${return_code}   0   msg=Fail to label worker node with instance-type=on-demand
+
+Teardown Kueue E2E Test Suite
+    [Documentation]    Teardown Kueue E2E Test Suite
+    Log To Console    Uninstall Kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl delete -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=Error detected while uninstalling kueue
+
+    # Remove label instance-type=on-demand from worker node
+    ${return_code} =    Run And Return Rc    oc label ${WORKER_NODE} instance-type-
+    Should Be Equal As Integers  ${return_code}   0   msg=Fail to unlabel instance-type=on-demand from worker node
+
+Run Kueue E2E Test
+    [Documentation]    Run Kueue E2E Test
+    [Arguments]    ${test_name}
+    Log To Console    Running Kueue E2E test: ${test_name}
+    ${return_code}    ${output}    Run And Return Rc And Output    ginkgo --focus-file=${test_name} ${KUEUE_DIR}/test/e2e/singlecluster
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=${test_name} failed
+
+Enable Visibility Feature Gate
+    [Documentation]    Enable Visibility Feature Gate
+    ${return_code} =    Run And Return Rc    oc patch deployment kueue-controller-manager -n kueue-system --type 'json' -p '[{"op" : "add" ,"path" : "/spec/template/spec/containers/0/args/-" ,"value" : "--feature-gates=VisibilityOnDemand=true"}]'
+    Should Be Equal As Integers  ${return_code}   0   msg=Visiblity feature gate is not enabled
```

Comment:
When I ran the file locally then some tests failed on HTTP 503. It looks like the problem was with Kueue still being restarted when tests were executed.
This helped me locally:
```
    ${return_code} =    Run And Return Rc    oc rollout status deployment/kueue-controller-manager -n kueue-system --timeout=3m
    Should Be Equal As Integers  ${return_code}   0   msg=Kueue rollout failed
```